aten/src/ATen/native/vulkan/api/Context.cpp

```diff
@@ -41,12 +41,11 @@ VkDevice create_device(
       &device_extension_properties_count,
       device_extension_properties.data()));
 
-  constexpr const char* const requested_device_extensions[]{
+  std::vector<const char*> requested_device_extensions;
```

Comment:
Not sure why this is a problem on C++14, especially since we are compiling on Clang and GCC with C++14 enabled, but anyhow if this doesn't compile for you please consider removing the constexpr and using a static array instead of std::vector.  Although this code is not performance sensitive there's no need for a dynamic allocation.

Reply:
Yeah sorry for the confusion. I'm mainly targeting  #54971 to make the code work with MSVC, which infamously has weaker support for `constexpr`. Will fix this with a static array later.